src/Storages/VirtualColumnUtils.cpp

```diff
@@ -193,6 +224,7 @@ void addRequestedFileLikeStorageVirtualsToChunk(
     Chunk & chunk, const NamesAndTypesList & requested_virtual_columns,
     VirtualsForFileLikeStorage virtual_values)
 {
+    auto hive_map = parseHivePartitioningKeysAndValues(virtual_values.hive_partitioning_path);
```

Comment:
Maybe we can propogate context to `addRequestedFileLikeStorageVirtualsToChunk` and check `use_hive_partitioning` here to avoid checking it in every storage.